api/pkg/s3/datastore/ceph/ceph.go

```diff
@@ -141,6 +141,10 @@ func (ad *CephAdapter) DELETE(object *pb.DeleteObjectInput, ctx context.Context)
 	return NoError
 }
 
+func (ad *CephAdapter) RestoreObject(object *pb.RestoreObjectInput, ctx context.Context) S3Error {
+	return NoError
```

Comment:
please return Not Supported wherever this operation is not supported (CEPH, GCP)